src/core_modules/capture-core/trackerOffline/shouldDiscard.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+import type { ApiError } from '@dhis2/app-runtime';
+
+const errorTypes = {
+    NETWORK: 'network',
+    ACCESS: 'access',
+    UNKNOWN: 'unknown',
+};
+
+const shouldDiscardRequestWithAccessError = ({ details: { httpStatusCode } }) => httpStatusCode !== 401;
+
+const shouldDiscardRequestWithUnknownError = ({ message }) => {
+    const statusCodeSearchResult = message.match(/\(\d*\)$/);
+    if (statusCodeSearchResult) {
+        const statusCode = statusCodeSearchResult[0].replace(/[()]/g, '');
+        return Boolean(statusCode) && ![408, 429, 502, 503, 504].includes(statusCode);
+    }
+    return false;
+};
+
+export const shouldDiscard = (error?: ApiError) => {
+    if (!error || !error.type) {
```

Comment:
just curious if you could do something like
```suggestion
    if (!error?.type) {
```

Reply:
Nice! That should work.